### ISSUE TITLE
training/validation/test splits

### ISSUE THREAD

**Issue by paalpri** [rpautrat/SuperPoint#140]
Hi! 

I noticed that in all my data splits i get the same images, and that's how it seems in the implementation as well, since the images are read from train2014 in every split. Is this on purpose or did i miss understand something maybe ?

**Comment by rpautrat** [comment]
Hi!

Yes this is not very good practice, but currently the network is trained on the full train2014, the validation set is a small subset of train2014 and there is no test set. Or rather, the test set option will return the full training set again, but we don't use test sets on COCO anyway, since HPatches is our test set in practice.

If you want to improve this, you just need to modify the file `coco.py` and consider three cases: whether it is training, validation or test.
